Modules/_pickle.c

```diff
@@ -4020,7 +4024,28 @@ save(PicklerObject *self, PyObject *obj, int pers_save)
         status = save_tuple(self, obj);
         goto done;
     }
-    else if (type == &PyType_Type) {
+    /*  The switch-on-type statement ends here because the next three
+     *  conditions are not exclusive anymore. If global_hook returns
+     *  NotImplementedError, then we must fallback to save_type or save_global
+     *  */
+    if (self->global_hook != NULL){
+        PyObject *reduce_value = NULL;
+        reduce_value = PyObject_CallFunctionObjArgs(self->global_hook, self, obj,
+                                                    NULL);
+        if (reduce_value == NULL){
+            goto error;
+        }
+
+        if (reduce_value != PyExc_NotImplementedError){
```

Comment:
I believe we should check for `builtins.NotImplement` (the singleton) instead of the `NotImplementedError` exception class (that is meant to be raised and not used as a return value).

Reply:
I do not know how to grab `builtins.NotImplement` using the Python C-API though.